types/vote_set.go

```diff
@@ -445,6 +445,26 @@ func (voteSet *VoteSet) StringIndented(indent string) string {
 		indent)
 }
 
+// Marshal the VoteSet to JSON. Same as String(), just in JSON,
+// and without the height/round/type_ (since its already included in the votes).
+func (voteSet *VoteSet) MarshalJSON() ([]byte, error) {
+	voteStrings := make([]string, len(voteSet.votes))
+	for i, vote := range voteSet.votes {
```

Comment:
what about 

```
voteSet.mtx.Lock()
defer voteSet.mtx.Unlock()
```

Reply:
probably right. String() doesn't have it tho